src/components/appointments/CreateAppointment.js

```diff
@@ -0,0 +1,202 @@
+import Types from 'types';
+
+import ChooseProductStep, {isStepValid as isProductStepValid} from './ChooseProductStep';
+import LocationAndTimeStep, {
+  isStepValid as isLocationAndTimeStepValid,
+} from './LocationAndTimeStep';
+
+const APPOINTMENT_STEPS = [
+  {
+    path: 'producten',
+    element: <ChooseProductStep />,
+    name: defineMessage({
+      description: "Appointments navbar title for '' step",
+      defaultMessage: 'Product',
+    }),
+    validCheck: isProductStepValid,
+  },
+  {
+    path: 'kalender',
+    element: <LocationAndTimeStep />,
+    name: defineMessage({
+      description: "Appointments navbar title for '' step",
+      defaultMessage: 'Location and time',
+    }),
+    validCheck: isLocationAndTimeStepValid,
+  },
+  {
+    path: 'contactgegevens',
+    element: <>Contactgegevens</>,
+    name: defineMessage({
+      description: "Appointments navbar title for '' step",
+      defaultMessage: 'Contact details',
+    }),
+    validCheck: () => false,
+  },
+];
+
+const APPOINTMENT_STEP_PATHS = APPOINTMENT_STEPS.map(s => s.path);
+
+/**
+ * Route subtree for appointment forms.
+ */
+export const routes = [
+  {
+    path: '',
+    element: <Navigate replace to={APPOINTMENT_STEP_PATHS[0]} />,
+  },
+  ...APPOINTMENT_STEPS.map(({path, element}) => ({path, element})),
+];
+
+/**
+ * Check if a given relative path from the routes matches the current location.
+ * @param currentPathname The current router location.pathname, from useLocation.
+ * @param path            The relative path to check for matches
+ */
+const checkMatchesPath = (currentPathname, path) => {
+  // we need to transform the path into a parent-route lookup, instead of using the
+  // default relative ./<foo> behaviour. The idea is that this component is mounted
+  // somewhere in a larger route definition but the exact parent route is not relevant.
+  const resolvedPath = resolvePath(`../${path}`, currentPathname);
+  // if the relative path is not the current URL, matchPath returns null, otherwise
+  // a match object.
+  const match = matchPath(resolvedPath.pathname, currentPathname);
+  return match !== null;
+};
+
+const AppointmentProgress = ({title, currentStep}) => {
+  const config = useContext(ConfigContext);
+  const {values} = useFormikContext();
+  const intl = useIntl();
+  const {pathname: currentPathname} = useLocation();
+
+  const [expanded, setExpanded] = useState(false);
+
+  const currentStepIndex = APPOINTMENT_STEP_PATHS.indexOf(currentStep);
+  const steps = APPOINTMENT_STEPS.map(({path, name, validCheck}) => {
+    const index = APPOINTMENT_STEP_PATHS.indexOf(path);
+    const previousStepIndex = Math.max(index - 1, 0);
+    const previousStepCompleted = APPOINTMENT_STEPS[previousStepIndex].validCheck(values);
+    const isValid = validCheck(values);
+    return {
+      uuid: `appointments-${path}`,
+      href: path,
+      isCompleted: previousStepCompleted && isValid,
```

Comment:
I noticed in storybook that now when you select the product, the step in the progress indicator automatically gets checked (marked as complete). This is a bit inconsistent with how it is in the form?

Reply:
I tried tweaking this around but it was odd either way - we don't really have an explicit submit event because all the state is just kept in memory and not synced to the backend (yet).

Let's discuss this after the 2.2 release